homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -0,0 +1,172 @@
+"""Train information for departures and delays, provided by Trafikverket."""
+
+import asyncio
+from datetime import date, datetime, timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_WEEKDAY, WEEKDAYS
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRAINS = "trains"
+CONF_FROM = "from"
+CONF_TO = "to"
+CONF_TIME = "time"
+
+ATTR_CANCELED = "canceled"
+ATTR_DELAY_TIME = "number_of_minutes_delayed"
+ATTR_PLANNED_TIME = "planned_time"
+ATTR_ESTIMATED_TIME = "estimated_time"
+ATTR_ACTUAL_TIME = "actual_time"
+ATTR_OTHER_INFORMATION = "other_information"
+ATTR_DEVIATIONS = "deviations"
+
+ICON = "mdi:train"
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_TRAINS): [{
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_TO): cv.string,
+        vol.Required(CONF_FROM): cv.string,
+        vol.Optional(CONF_TIME): cv.time,
+        vol.Optional(CONF_WEEKDAY, default=WEEKDAYS):
+            vol.All(cv.ensure_list, [vol.In(WEEKDAYS)])}]
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(
+        hass, config, async_add_devices, discovery_info=None):
+    """Set up the departure sensor."""
+    from pytrafikverket import TrafikverketTrain
+    httpsession = async_get_clientsession(hass)
+    train_api = TrafikverketTrain(httpsession, config.get(CONF_API_KEY))
+    sensors = []
+    for train in config.get(CONF_TRAINS):
+        try:
+            from_station = yield from train_api.async_get_train_station(
+                train.get(CONF_FROM))
+            to_station = yield from train_api.async_get_train_station(
+                train.get(CONF_TO))
+        except ValueError as station_error:
+            _LOGGER.error("Problem when trying station %s to %s. Error: %s ",
+                          train.get(CONF_FROM), train.get(CONF_TO),
+                          station_error)
+            return
```

Comment:
By returning, other valid sensors are not added either.

Reply:
Added check to see if the error is due to invalid authentication (api_key) then it is no point trying to add more. 
Otherwise it would continue and move next in the loop.